src/isp_emitters.ml

```diff
@@ -271,46 +271,84 @@ module Auxiliary = struct
       p_warning
         "The term %a is a pointer! Eva can't evaluate this, and thus no \
          annotations are created for this term."
-        Printer.pp_term t
+        Printer.pp_term term
+
+  (** Add 'spec_type' annotations for the given lval to the infered behavior
+      contract of the given function. 'to_term' is a function that converts
+      an lval into a term. *)
+  let emit_lval_spec spec_type lvalue to_term req new_kf filling_actions =
+    let rec find_field_offsets typ =
+      match Cil.unrollType typ with
+      | TNamed _ -> 
+        (* TODO: May be the case with TPtr TArray etc. Check Cil.unrollTypeDeep. *)
+        failwith "Trying to emit annotations for non-unrolled type."
+      | TComp (compinfo, _) ->
+          List.flatten 
+            (List.map
+              (fun fieldinfo ->
+                let o = find_field_offsets fieldinfo.ftype in
+                List.map (fun f -> Field (fieldinfo, f)) o)
+              (Option.value compinfo.cfields ~default:[]))
+      | _ -> [NoOffset]
+    in
+    match Cil.unrollType (Cil.typeOfLval lvalue) with
+    | TNamed _ -> 
+      (* TODO: May be the case with TPtr TArray etc. Check Cil.unrollTypeDeep. *)
+      failwith "Trying to emit annotations for non-unrolled type."
+    | TComp _ as styp->
+        let (lhost, _) = lvalue in
+        let offsets = find_field_offsets styp in
+        p_debug "···· number of found offsets %i" (List.length offsets) ~level:4;
+        List.iter 
+          (fun o ->
```

Comment:
Sure. (What is the naming strategy? I looked around and found different things.)

Reply:
There is no "global" naming strategy, I meant more locally to the function. Since you have used the name e.g. `offsets`, then I feel it would be natural to use `offset` in this situation.